examples/http/vmware_vcenter_monitoring_system_health.js

```diff
@@ -0,0 +1,159 @@
+/**
+ * Domotz Custom Driver 
+ * Name: VMWare vCenter Monitoring System Health
+ * Description: This script is designed to retrieve system health information from a VMware vCenter server
+ *               
+ * Communication protocol is HTTPS
+ * 
+ * Tested on VMWare vCenter version 8.0.2
+ *
+ * Creates a Custom Driver table with the system health information
+ *
+ **/
+
+// Create a Custom Driver table to store system health information
+var table = D.createTable(
+    "System Health", [
+        { label: "Status", valueType: D.valueType.STRING }
+    ]
+);
+
+// Function to login to the VMWare vCenter
+function login() {
+    var d = D.q.defer();
+    var config = {
+        url: "/api/session",
+        username: D.device.username(),
+        password: D.device.password(),
+        protocol: "https",
+        auth: "basic",
+        jar: true,
+        rejectUnauthorized: false
+    };
+    D.device.http.post(config, function(error, response, body){
+        if (error) {          
+            console.error(error);
+            D.failure(D.errorType.GENERIC_ERROR);
+        } else if (response.statusCode == 404) {
+            D.failure(D.errorType.RESOURCE_UNAVAILABLE);
+        } else if (response.statusCode == 401) {
+            D.failure(D.errorType.AUTHENTICATION_ERROR);
+        } else if (response.statusCode != 201) {
+            D.failure(D.errorType.GENERIC_ERROR);
+        } 
+        d.resolve(JSON.parse(body));
+    });
+    return d.promise;
+}
+
+/**
+ * Function to perform HTTP GET request
+ * @param {string} url The URL to send the GET request to
+ * @param {object} sessionId The response from the login request, containing the session ID
+ * @returns A promise containing the body of the response
+ */
+function httpGet(url, sessionId) {
+    var d = D.q.defer();
+    var config = {
+        url: url,
+        protocol: "https",
+        jar: true,
+        rejectUnauthorized: false,
+        headers: {
+            "vmware-api-session-id": sessionId
+        }
+    };
+    D.device.http.get(config, function(error, response, body){
+        if (error) {          
+            console.error(error);
+            D.failure(D.errorType.GENERIC_ERROR);
+        } else if (response.statusCode == 404) {
+            D.failure(D.errorType.RESOURCE_UNAVAILABLE);
+        } else if (response.statusCode != 200) {
+            console.log(response.body);
+            d.resolve("N/A");
+        } 
+        d.resolve(JSON.parse(body));
+    });
+    return d.promise;
+}
+
+// Function to get URLs for system health information
+function getHealthSystemsUrls(url) {
```

Comment:
there is no need for this function, you can use directly httpGet(url, sessionId)